test-app/runtime/src/main/cpp/modules/url/URL.cpp

```diff
@@ -0,0 +1,343 @@
+}
+
+napi_value URL::SetHost(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_host);
+}
+
+napi_value URL::SetHostName(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_hostname);
+}
+
+napi_value URL::SetHref(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_href);
+}
+
+napi_value URL::SetPassword(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_password);
+}
+
+napi_value URL::SetPathName(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_pathname);
+}
+
+napi_value URL::SetPort(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_port);
+}
+
+napi_value URL::SetProtocol(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_protocol);
+}
+
+napi_value URL::SetSearch(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_search);
+}
+
+napi_value URL::SetUserName(napi_env env, napi_callback_info info) {
+    return SetUrlProperty(env, info, &url_aggregator::set_username);
+}
+
+
+// Add toString method
+napi_value URL::ToString(napi_env env, napi_callback_info info) {
+    NAPI_PREAMBLE
+    URL *instance = GetInstance(env, info);
+    if (!instance)
+        return nullptr;
+
+    auto value = instance->GetURL()->get_href();
+
+    napi_value result;
+    NAPI_GUARD(napi_create_string_utf8(env, value.data(), value.length(), &result)) {
+        return nullptr;
+    }
+
+    return result;
+}
+
+// Constructor
+napi_value URL::New(napi_env env, napi_callback_info info) {
+    NAPI_CALLBACK_BEGIN(2)
+
+    if (argc < 1) {
+        napi_throw_type_error(env, nullptr, "URL constructor requires at least 1 argument");
+        return nullptr;
+    }
+
+    size_t str_size;
+    NAPI_GUARD(napi_get_value_string_utf8(env, argv[0], nullptr, 0, &str_size)) {
+        return nullptr;
+    }
+
+    std::vector<char> url_buffer(str_size + 1);
+    NAPI_GUARD(napi_get_value_string_utf8(env, argv[0], url_buffer.data(), str_size + 1, nullptr)) {
+        return nullptr;
+    }
+
+    url_aggregator url;
+    std::string url_string(url_buffer.data());
+
+    if (argc > 1) {
+        // Handle base URL
+        size_t base_str_size;
+        NAPI_GUARD(napi_get_value_string_utf8(env, argv[1], nullptr, 0, &base_str_size)) {
+            return nullptr;
+        }
+
+        std::vector<char> base_buffer(base_str_size + 1);
+        NAPI_GUARD(napi_get_value_string_utf8(env, argv[1], base_buffer.data(), base_str_size + 1,
+                                              nullptr)) {
+            return nullptr;
+        }
+
+        std::string base_string(base_buffer.data());
+        std::string_view base_string_view(base_string);
```

Comment:
Previously you were creating an unnecessary string

```suggestion
        std::string_view base_string_view(base_buffer.data(), base_buffer.size());
```